mirgecom/boundary.py

```diff
@@ -38,91 +40,264 @@
+        """Get the viscous boundary flux for the divergence operator."""
+
+    @abstractmethod
+    def cv_gradient_flux(self, discr, btag, gas_model, state_minus, **kwargs):
+        """Get the fluid soln boundary flux for the gradient operator."""
+
+    @abstractmethod
+    def temperature_gradient_flux(self, discr, btag, gas_model, state_minus,
+                                  **kwargs):
+        r"""Get temperature flux across the boundary faces."""
+
 
 class PrescribedFluidBoundary(FluidBoundary):
     r"""Abstract interface to a prescribed fluid boundary treatment.
 
     .. automethod:: __init__
     .. automethod:: inviscid_divergence_flux
+    .. automethod:: cv_gradient_flux
+    .. automethod:: viscous_divergence_flux
+    .. automethod:: temperature_gradient_flux
     """
 
     def __init__(self,
                  # returns the flux to be used in div op (prescribed flux)
-                 inviscid_boundary_flux_func=None,
+                 inviscid_flux_func=None,
                  # returns CV+, to be used in num flux func (prescribed soln)
                  boundary_state_func=None,
                  # Inviscid facial flux func given CV(+/-)
-                 inviscid_facial_flux_func=None,
+                 # inviscid_numerical_flux_func=None,
                  # Flux to be used in grad(Temperature) op
                  temperature_gradient_flux_func=None,
                  # Function returns boundary temperature_plus
-                 boundary_temperature_func=None):
+                 boundary_temperature_func=None,
+                 # Function returns the flux to be used in grad(cv)
+                 cv_gradient_flux_func=None,
+                 # Function computes the numerical flux for a gradient
+                 gradient_numerical_flux_func=None,
+                 # Function computes the flux to be used in the div op
+                 viscous_flux_func=None,
+                 # Returns the boundary value for grad(cv)
+                 boundary_gradient_cv_func=None,
+                 # Returns the boundary value for grad(temperature)
+                 boundary_gradient_temperature_func=None,
+                 ):
         """Initialize the PrescribedFluidBoundary and methods."""
         self._bnd_state_func = boundary_state_func
-        self._inviscid_bnd_flux_func = inviscid_boundary_flux_func
-        self._inviscid_div_flux_func = inviscid_facial_flux_func
+        self._inviscid_flux_func = inviscid_flux_func
+        self._temperature_grad_flux_func = temperature_gradient_flux_func
         self._bnd_temperature_func = boundary_temperature_func
+        self._grad_num_flux_func = gradient_numerical_flux_func
+        self._cv_gradient_flux_func = cv_gradient_flux_func
+        self._viscous_flux_func = viscous_flux_func
+        self._bnd_grad_cv_func = boundary_gradient_cv_func
+        self._bnd_grad_temperature_func = boundary_gradient_temperature_func
```

Comment:
A struct of callback functions is not lovely: Use an abstract base class. There's even (sort of, hand-coded) inheritance logic below!

Reply:
We are aware of this issue and plan to refactor the boundary treatments accordingly - based on previous feedback from @inducer and @majosm.   Is that something we want to tackle in this PR?

Reply:
Bump to issue is OK with me.

Reply:
#576 is the issue created for this around the time we last discussed it.